File: evals/refs/pal.zig
const std = @import("std");

pub fn main(init: std.process.Init) !void {
    var in_buf: [4096]u8 = undefined;
    var fr = std.Io.File.stdin().reader(init.io, &in_buf);
    const r = &fr.interface;

    var out_buf: [4096]u8 = undefined;
    var fw = std.Io.File.stdout().writer(init.io, &out_buf);
    const w = &fw.interface;

    var clean: [4096]u8 = undefined;

    while (try r.takeDelimiter('\n')) |line| {
        var n: usize = 0;
        for (line) |c| {
            if (std.ascii.isAlphanumeric(c)) {
                clean[n] = std.ascii.toLower(c);
                n += 1;
            }
        }
        var is_pal = true;
        var i: usize = 0;
        while (i < n / 2) : (i += 1) {
            if (clean[i] != clean[n - 1 - i]) {
                is_pal = false;
                break;
            }
        }
        try w.writeAll(if (is_pal) "true\n" else "false\n");
    }
    try w.flush();
}

File: evals/refs/pal.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"unicode"
)

func isPalindrome(line string) bool {
	var clean []rune
	for _, r := range line {
		if unicode.IsLetter(r) || unicode.IsDigit(r) {
			clean = append(clean, unicode.ToLower(r))
		}
	}
	for i, j := 0, len(clean)-1; i < j; i, j = i+1, j-1 {
		if clean[i] != clean[j] {
			return false
		}
	}
	return true
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Buffer(make([]byte, 1024*1024), 1024*1024)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	for scanner.Scan() {
		if isPalindrome(scanner.Text()) {
			fmt.Fprintln(writer, "true")
		} else {
			fmt.Fprintln(writer, "false")
		}
	}
}
